### ISSUE TITLE
Dependency injection example using unity container in .net framework 4

### ISSUE THREAD

**Issue by clustersnake** [graphql-dotnet/graphql-dotnet#843]
## Summary
I'm currently using .net framework and I can't find a working example of dependency injection with unity container, in fact all I've found are .net core examples but I'm stuck with the old platform.
Could you please give some indications about the correct use of dependency injection in GraphQL types? I had to instantiate my repositories inside my types to make the queries and It just doesn't feel right.
## Relevant information
I'm using unity container for my repositories dependency injection in the web controller
https://github.com/unitycontainer/unity

and I'm configuring it in WebApiConfig.cs

```
    public static class WebApiConfig
    {
        public static void Register(HttpConfiguration config)
        {
            // Web API configuration and services
            var container = new UnityContainer();
            container.RegisterType<IUsuarioRepository, UsuarioRepository>(new HierarchicalLifetimeManager());
            container.RegisterType<IGrupoRepository, GrupoRepository>(new HierarchicalLifetimeManager());
            container.RegisterType<ITareaRepository, TareaRepository>(new HierarchicalLifetimeManager());

            config.DependencyResolver = new UnityResolver(container);


```
I really don't know if it's possible to inject the repositories in my types constructors to be able to make the queries using them

### Environment (if relevant)
.net framework 4.6.1
webApi 5.2

**Comment by joro550** [comment]
Yes this is possible, but it is kind of hidden away in the documentation, let me point it out so it may become clearer: https://graphql-dotnet.github.io/docs/getting-started/dependency-injection
This page talks about dependency injection in your query typs if you look very closely you'll see

**Comment by clustersnake** [comment]
@joro550 
Wow! I was working on the frontend tasks and I just forgot about this for a little. I'm going to try it right now.Thank you very much!

**Comment by clustersnake** [comment]
@joro550 maybe I'm just too noob to this. I have to really learn this
![image](https://user-images.githubusercontent.com/16407149/47228712-e1fe6900-d393-11e8-8351-a6ba43ac9a5b.png)

**Comment by joemcbride** [comment]
It may be easier to implement the container directly.  Something like:

```csharp
public class UnityDependencyResolver : IDependencyResolver
{
    private readonly IUnityContainer _container;

    public UnityDependencyResolver(IUnityContainer container)
    {
        _container = container;
    }

    public T Resolve<T>()
    {
        return (T)Resolve(typeof(T));
    }

    public object Resolve(Type type)
    {
        return _container.Resolve(type);
    }
}
```

**Comment by clustersnake** [comment]
@joemcbride thanks, I have to learn DI in detail. I'm actually playing the one man band role and since the app is running I have to move on without it :-/
